torchmultimodal/modules/layers/attention.py

```diff
@@ -0,0 +1,240 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Dict, Optional, Tuple
+
+import torch
+from torch import nn, Tensor
+from torch.nn import functional as F
+from torchmultimodal.utils.common import shift_dim
+
+
+class MultiHeadAttention(nn.Module):
+    """Compute multihead attention with flexible attention function.
+
+    Multihead attention linearly projects queries, keys, and values into an embedding
```

Comment:
`embedding space` sounds very generic. Would instead focus on the purpose of the linear projection which is to run the scaled product attention multiple times in parallel.